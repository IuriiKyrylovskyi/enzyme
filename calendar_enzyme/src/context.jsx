import React, { useState, useContext } from "react";
import moment from "moment";

const AppContext = React.createContext();

const AppProvider = ({ children }) => {
  const [isOpen, setIsOpen] = useState(false);
  const [isEvent, setEvent] = useState(false);

  const [dateInput, setDateInput] = useState("");
  const [startTimeInput, setStartTimeInput] = useState("");
  const [endTimeInput, setEndTimeInput] = useState("");

  const onCloseModal = () => {
    setIsOpen(false);
  };

  const onOpenModal = (e) => {
    if (isEvent || e.target.closest(".event")) {
      return; // setEvent(false);
    }
    if (e.target.className === "calendar__time-slot") {
      // console.log(onOpenModalStartTime());
      setDateInput("");
      setStartTimeInput("");
      setEndTimeInput("");

      setEvent(false);
      setIsOpen(true);
      console.log(e);
      console.log(e.nativeEvent);
      return;
    }

    if (!e.target.closest("event")) {
      console.log(e.target.className);
      setDateInput(moment().format("YYYY-MM-DD"));
      setStartTimeInput(moment().format("HH:mm"));
      setEndTimeInput(moment().format("HH:mm"));

      setEvent(false);
      setIsOpen(true);
    }
  };

  // function onOpenModalStartTime(start) {
  //   setStartTimeInput(start);
  // }

  return (
    <AppContext.Provider
      value={{
        isOpen,
        isEvent,
        onCloseModal,
        onOpenModal,

        dateInput,
        startTimeInput,
        endTimeInput,
        // onOpenModalStartTime,
      }}
    >
      {children}
    </AppContext.Provider>
  );
};

export const useGlobalContext = () => {
  return useContext(AppContext);
};

export { AppProvider };
