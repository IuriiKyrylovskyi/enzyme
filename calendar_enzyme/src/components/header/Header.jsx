import React from "react";
import { useGlobalContext } from "../../context";
// import Modal from "../modal/Modal";

// import moment from 'moment';

import { months } from "../../utils/dateUtils.js";

import "./header.scss";

const Header = (props) => {
  const { isOpen, onOpenModal, onCloseModal } = useGlobalContext();
  const { weekDates, handleTodayBtn, handleArrowBtn } = props;
  // const prevMonth = moment(props.weekDates[0]).format('MMM');
  // const nextMonth = moment(props.weekDates[6]).format('MMM');

  const monthAtWeekStart = months[new Date(weekDates[0]).getMonth()];
  const monthAtWeekEnd = months[new Date(weekDates[6]).getMonth()];

  return (
    <header className="header">
      <button className="button create-event-btn" onClick={onOpenModal}>
        <i className="fas fa-plus create-event-btn__icon"></i>Create
      </button>
      <div className="navigation">
        <button className="navigation__today-btn button" onClick={handleTodayBtn}>
          Today
        </button>
        <button className="icon-button navigation__nav-icon" onClick={() => handleArrowBtn(-7)}>
          <i className="fas fa-chevron-left"></i>
        </button>
        <button className="icon-button navigation__nav-icon" onClick={() => handleArrowBtn(7)}>
          <i className="fas fa-chevron-right"></i>
        </button>
        <span className="navigation__displayed-month">{`${monthAtWeekStart === monthAtWeekEnd ? monthAtWeekStart : monthAtWeekStart + " - " + monthAtWeekEnd}`}</span>
      </div>
      {/* {isOpen && <Modal isOpen={isOpen} onCloseModal={onCloseModal} />} */}
    </header>
  );
};

export default Header;
