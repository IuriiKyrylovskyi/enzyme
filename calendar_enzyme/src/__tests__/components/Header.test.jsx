import React from "react";
import { shallow, mount } from "enzyme";
import { AppProvider } from "../../context";
import Header from "../../components/header/Header";

const props = {
  weekDates: [],
  handleTodayBtn: jest.fn(),
  handleArrowBtn: jest.fn(),
  onOpenModal: jest.fn(),
};

describe("<Header />", () => {
  const wrappedComponent = mount(
    <AppProvider>
      <Header {...props} />
    </AppProvider>
  );

  it("should display component", () => {
    expect(wrappedComponent).toMatchSnapshot();
  });

  // it("should open modal on button click", () => {
  //   wrappedComponent.find(".create-event-btn").prop("onClick")();

  //   expect(props.onOpenModal).toBeCalled();
  // });

  it("should display 'Create'", () => {
    expect(wrappedComponent.find(".create-event-btn").text()).toEqual("Create");
  });

  it("should display current week on button click", () => {
    wrappedComponent.find(".navigation__today-btn").prop("onClick")();

    expect(props.handleTodayBtn).toBeCalled();
  });

  it("should display 'Today'", () => {
    expect(wrappedComponent.find(".navigation__today-btn").text()).toEqual("Today");
  });

  it("should display previos week on button click", () => {
    wrappedComponent.find(".icon-button").at(0).prop("onClick")();

    expect(props.handleArrowBtn).toBeCalled();
  });

  it("should display next week on button click", () => {
    wrappedComponent.find(".icon-button").at(1).prop("onClick")();

    expect(props.handleArrowBtn).toBeCalled();
  });

  it("should display month", () => {
    expect(wrappedComponent.find(".navigation__displayed-month").exists()).toBeTruthy();
  });
});
