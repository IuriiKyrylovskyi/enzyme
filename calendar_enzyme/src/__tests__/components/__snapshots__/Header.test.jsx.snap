// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should display component 1`] = `
<AppProvider>
  <Header
    handleArrowBtn={[MockFunction]}
    handleTodayBtn={[MockFunction]}
    onOpenModal={[MockFunction]}
    weekDates={Array []}
  >
    <header
      className="header"
    >
      <button
        className="button create-event-btn"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus create-event-btn__icon"
        />
        Create
      </button>
      <div
        className="navigation"
      >
        <button
          className="navigation__today-btn button"
          onClick={[MockFunction]}
        >
          Today
        </button>
        <button
          className="icon-button navigation__nav-icon"
          onClick={[Function]}
        >
          <i
            className="fas fa-chevron-left"
          />
        </button>
        <button
          className="icon-button navigation__nav-icon"
          onClick={[Function]}
        >
          <i
            className="fas fa-chevron-right"
          />
        </button>
        <span
          className="navigation__displayed-month"
        >
          undefined
        </span>
      </div>
    </header>
  </Header>
</AppProvider>
`;
